src/main/java/com/yahoo/bullet/rest/pubsub/PubSubService.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright 2017, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.rest.pubsub;
+
+import com.yahoo.bullet.pubsub.PubSubMessage;
+import org.springframework.stereotype.Service;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+@Service
+public class PubSubService {
+    private ConcurrentLinkedQueue<PubSubMessage> queries = new ConcurrentLinkedQueue<>();
```

Comment:
I think you may just store the json string instead of PubSubMessage object. Then you don't need to serialize and deserialize them every time you poll and push.